### PR TITLE
Enrich transactions with more data

### PR DESCRIPTION
**Pull request by Tyrrrz**

### CODE REVIEW

File: src/Sentry/Internal/Enricher.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            if (_sdkVersionLazy.Value.Version is not null)
+            {
+                eventLike.Sdk.AddPackage("nuget:" + _sdkVersionLazy.Value.Name, _sdkVersionLazy.Value.Version);
+            }
+
+            // Release
+            if (eventLike.Release is null)
+            {
+                eventLike.Release = _options.Release ?? _releaseLazy.Value;
+            }
```
Comment (bruno-garcia) [inline src/Sentry/Internal/Enricher.cs:71]: Can't it be?

```suggestion
            eventLike.Release ??= _options.Release ?? _releaseLazy.Value;
```

File: src/Sentry/Internal/Enricher.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+            // Environment
+            if (string.IsNullOrWhiteSpace(eventLike.Environment))
+            {
+                var foundEnvironment = EnvironmentLocator.Locate();
```
Comment (bruno-garcia) [inline src/Sentry/Internal/Enricher.cs:76]: Can't recall if `EnvironmentLocator.Locate();` caches the result but it should

File: src/Sentry/Internal/Hub.cs [hunk 3]
```diff
@@ -111,31 +111,8 @@ public void WithScope(Action<Scope> scopeCallback)
...
-
-            // Environment information
-            var foundEnvironment = EnvironmentLocator.Locate();
-            transaction.Environment ??= string.IsNullOrWhiteSpace(foundEnvironment)
-                ? string.IsNullOrWhiteSpace(_options.Environment)
-                    ? Constants.ProductionEnvironmentSetting
-                    : _options.Environment
-                : foundEnvironment;
+            // Apply enricher
+            new Enricher(_options).Apply(transaction);
```
Comment (bruno-garcia) [inline src/Sentry/Internal/Hub.cs:115]: Why do we get a new instance of Enricher for each event? cuz draft PR?

**Review by bruno-garcia — COMMENTED** [review]
Going to a great direction!

File: src/Sentry/Internal/Enricher.cs (hunk 2 above)
Comment (Tyrrrz) [reply src/Sentry/Internal/Enricher.cs:73]: Actually, `Locate()` apparently doesn't. `Current` does. Changed.

**Comment by Tyrrrz** [comment]
Sample:

image

**Review by bruno-garcia — APPROVED** [review]

**Comment by Tyrrrz** [comment]
Suddenly failing consistently now, but only on Windows:

```
Error: Sentry.Tests.Internals.Http.CachingTransportTests.DoesNotDeleteCacheIfConnectionWithIssue: Test execution timed out after 7000 milliseconds
  Failed Sentry.Tests.Internals.Http.CachingTransportTests.DoesNotDeleteCacheIfConnectionWithIssue [1 ms]
  Error Message:
   Test execution timed out after 7000 milliseconds
```

**Pull request merged**